lib/appSearch.js

```diff
@@ -43,27 +43,11 @@ class AppSearchClient {
     this.client = new Client(apiKey, baseUrl)
   }
 
-  /**
-   * Send a search request to the App Search Api
-   * https://swiftype.com/documentation/app-search/api/overview
-   *
-   * @param {String} engineName unique Engine name
-   * @param {String} query String that is used to perform a search request.
-   * @param {Object} options Object used for configuring the search like search_fields and result_fields
-   * @returns {Promise<Object>} a Promise that returns a result {Object} when resolved, otherwise throws an Error.
-   */
```

Comment:
Its kind of a bummer to lose these helpful code comments

Reply:
Which ones, the parameter comments? I could add them to the definition files.

Reply:
And are you thinking about comments for a consumer of this library, or for developers?

Reply:
Yeah, maybe with some discretion? I'm sure a developer can probably guess what `engineName: string` is used for haha.  On the other hand, maybe types make this sort of thing "self documenting"? I'd handle in a follow-up either way, that work shouldn't block this PR in my opinion

Reply:
> And are you thinking about comments for a consumer of this library, or for developers?

A customer/developer integrating `app-search-node` into their app.  

Looking into this a little, I think you should be able to copy/paste them into our typecript files, I get the impression `tsdoc` and `jsdoc` share syntax.  See https://tsdoc.org/

Reply:
I think you're right. And that makes sense, I'll do this in a follow-up.

Reply:
@byronhulcher I just went ahead and re-added it: https://github.com/elastic/app-search-node/pull/50/commits/32b894e56d2eb2c441801384f31a55b06fe1fa75